Lib/interpreters.py

```diff
@@ -0,0 +1,240 @@
+"""Subinterpreters High Level Module."""
+
+import _interpreters
+import logger
+
+__all__ = ['Interpreter', 'SendChannel', 'RecvChannel', 'is_shareable',
+           'create_channel', 'list_all_channels', 'list_all', 'get_current',
+           'create']
+
+
+class Interpreter:
+
+    def __init__(self, id):
+        self.id = id
+
+    def is_running(self):
+        """is_running() -> bool
+
+        Return whether or not the identified interpreter is running.
+        """
+        return _interpreters.is_running(self.id)
+
+    def destroy(self):
+        """destroy()
+
+        Destroy the identified interpreter.
+
+        Attempting to destroy the current
+        interpreter results in a RuntimeError. So does an unrecognized ID
+        """
+        return _interpreters.destroy(self.id)
+
+    def run(self, src_str, /, *, channels=None):
+        """run(src_str, /, *, channels=None)
+
+        Run the given source code in the interpreter.
+        This blocks the current thread until done.
+        """
+        try:
+            _interpreters.run_string(self.id, src_str)
+        except RunFailedError as err:
+            logger.error(err)
+            raise
+
+def wait(self, timeout):
+    #The implementation for wait
+    # will be non trivial to be useful
+    import time
+    time.sleep(timeout)
+
+def associate_interp_to_channel(id, cid):
+    pass
+
+class RecvChannel:
+
+    def __init__(self, id):
+        self.id = id
+        self.interpreters = _interpreters.list_all()
+
+    def recv(self, timeout=2):
+        """ channel_recv() -> obj
+
+        Get the next object from the channel,
+        and wait if none have been sent.
+        Associate the interpreter with the channel.
+        """
+        obj = _interpreters.channel_recv(self.id)
+        if obj == None:
+            wait(timeout)
+            obj = obj = _interpreters.channel_recv(self.id)
```

Comment:
```suggestion
            obj = _interpreters.channel_recv(self.id)
```

Reply:
`channel_recv` doesn't return `None` when channel is empty, but rather it raises `ChannelEmptyError`. So I'd imagine this method try to read the channel, catch the error, and re-try. There are several possible approaches for this:
- it can be either poll (but channels don't support that, as of now), 
- we could divide the input timeout by some "magical" number and re-try every X amount of time.